newIDE/app/src/MainFrame/index.js

```diff
@@ -935,6 +936,8 @@ export default class MainFrame extends React.Component<Props, State> {
   save = () => {
     if (!this.state.currentProject) return;
 
+    saveUiSettings(this.state.editorTabs);
```

Comment:
So the Flow error is a bit tricky. In theory, Flow has no ideas what the heck you can be doing in saveUiSettings. Hence it assume that you could, inside this function, do some black magic and get a reference to the component, then to `this.state`, and change `this.state.currentProject` to be `null` or `undefined`. Unlikely, but possible.

So even if we have a check `if (!this.state.currentProject) return;` BEFORE saveUiSettings, Flow tells you that there is a chance that `this.state.currentProject` can be null.

Solution:
* either do `const {currentProject} = this.state;` and use currentProject in the rest of the function
* or move saveUiSettings before the `if (!this.state.currentProject) return;`

Reply:
Thanks :) One of the two solutions I suggested should work.